### PR TITLE
chore: add eslint

### PR DESCRIPTION
**Pull request by malept**
Follow-up to https://github.com/electron/get/pull/182#discussion_r561418583. It's mostly moving code around, adding/clarifying type annotations, and ignoring a bunch of `any`s :grimacing: 

Requires #183.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,17 +59,32 @@
...
   },
   "husky": {
     "hooks": {
       "pre-commit": "lint-staged"
     }
   },
+  "lint-staged": {
+    "*.ts": [
+      "eslint --fix",
+      "prettier --write",
```
Comment (erickzhao) [inline package.json:84]: Question: Is it necessary to have ESLint and Prettier running given that `eslint-config-prettier` is set up now?

Comment (ckerr) [reply package.json:84]: I'm not positive but I _think_ explicit invocation of both is still useful as-is in this PR.

If we wanted to combine the steps we could use `eslint-plugin-prettier`

File: test/utils.spec.ts [hunk 2]
```diff
@@ -96,24 +99,27 @@ describe('utils', () => {
...
       });
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
       process.config.variables = {} as any;
       expect(getHostArch()).toEqual('armv7l');
     });
 
-    it('should return uname on arm 6', () => {
-      if (process.platform !== 'win32') {
+    if (process.platform !== 'win32') {
+      it('should return uname on arm 6', () => {
```
Comment (ckerr) [inline test/utils.spec.ts:108]: :+1: on reversing this order

Just curious, was the linter smart enough to catch this, or did you spot it while adding the eslint-disable-next-line comments?

**Review by ckerr — APPROVED** [review]
LGTM pending confirmation on @erickzhao's question

File: package.json (hunk 1 above)
Comment (erickzhao) [reply package.json:84]: Ah, makes sense to me, I got `eslint-plugin-prettier` and `eslint-config-prettier` mixed up in my head. Thank you!

File: test/utils.spec.ts (hunk 2 above)
Comment (malept) [reply test/utils.spec.ts:108]: IIRC the linter (`eslint-plugin-jest`) was smart enough, it didn't like conditional `expect`s.

**Pull request merged**